src/DiIiS-NA/D3-GameServer/GSSystem/PlayerSystem/Player.cs

```diff
@@ -629,21 +629,37 @@ public void SetAttributesMisc()
 
 		public void SetAttributesByParagon()
 		{
+
+			// Set Paragon max 20.0000
+			int level = Math.Min(this.Toon.ParagonLevel, 20000);
+
 			this.Attributes[GameAttribute.Paragon_Bonus_Points_Available, 0] = 0;
 			this.Attributes[GameAttribute.Paragon_Bonus_Points_Available, 1] = 0;
 			this.Attributes[GameAttribute.Paragon_Bonus_Points_Available, 2] = 0;
 			this.Attributes[GameAttribute.Paragon_Bonus_Points_Available, 3] = 0;
-			int level = Math.Min(this.Toon.ParagonLevel, 800);
+
 			for (int i = 0; i < level; i++)
```

Comment:
This cycle can be simplified to 
```
    var basePoints = Math.Min(this.Toon.ParagonLevel, 800);
    var extraPoints = Math.Max(0, this.Toon.ParagonLevel - 800);
    for (int i = 0; i < 4; i++)
    {
        this.Attributes[GameAttribute.Paragon_Bonus_Points_Available, i] = basePoints / 4;
        this.Attributes[GameAttribute.Paragon_Bonus_Points_Available, i] += extraPoints / 4;
        if (i < this.Toon.ParagonLevel % 4)
        {
            this.Attributes[GameAttribute.Paragon_Bonus_Points_Available, i]++;
        }
    }
```

Reply:
I'll implemented and test that. 
I tried to keep the same structure and these are my first steps in C#, suggestions and tips are very welcome!!

Reply:
Yeah, on tests I got this issue and implemented something like this.

`			// Until the Paragon 800 should be distributed on the 4 tabs,
			// after that only in the first Core tab.
			var baseParagonPoints = Math.Min(this.Toon.ParagonLevel, 800);
			var extraParagonPoints = Math.Max(0, this.Toon.ParagonLevel - 800);
			for (int i = 0; i < 4; i++)
			{
				this.Attributes[GameAttribute.Paragon_Bonus_Points_Available, i] = baseParagonPoints / 4;
				// First tab Core. Pos 0 in array.
				this.Attributes[GameAttribute.Paragon_Bonus_Points_Available, 0] += extraParagonPoints / 4;
				if (i < this.Toon.ParagonLevel % 4)
				{
					this.Attributes[GameAttribute.Paragon_Bonus_Points_Available, i]++;
				}
			}`

Reply:
is it possible for me to remove the limit of adding 255 points at a time? After I changed ParagonBonuses to ushort I could go over 255, but the problem is that it only accepts to add 255 at a time and when you have a lot of paragon it is interesting to be able to add 1000 at a time or more.

Reply:
1) Set limit of Paragon to 20k instead 800.
2) Fixed Paragon's point distribution. After Level 800 only the Core Tab receives points.
3) Removed 255 maximum points limit for attribute of Paragon.
4) Changed add point limit once, now you can add 1k~20k in vitality at once and the field also accepts it.
These changes require deleting the current accounts, as I have changed the database field typing
Now the paragon it's ok.